### DISCUSSION TITLE
Create a Bitmap image from the live stream video being published

### DISCUSSION THREAD

**Discussion by AlexUrrutia** [HaishinKit/HaishinKit.swift#1312] [Idea]
Hello, I want to know if it is possible to create a Bitmap image from what is being published to the server to create a preview thumbnail, like the Pedro's library does: [RootEncoder Library](https://github.com/pedroSG94/RootEncoder/blob/52baa9aed4422dd0f8e12d451df80d16ee3b12c0/library/src/main/java/com/pedro/library/view/OpenGlViewBase.java#L144)

```
rtmpCamera1.getGlInterface().takePhoto(new TakePhotoCallback() {
          @Override
          public void onTakePhoto(Bitmap bitmap) {
            setImageToStream(bitmap);
          }
        });
```

**Comment by shogo4405** [comment]
There is no dedicated interface provided. You can access the image using VideoEffect.
https://shogo4405.github.io/HaishinKit.swift/documentation/haishinkit/videoeffect

**Reply by AlexUrrutia** [reply]
Oh I see, let me try it .. Thanks!

**Comment by AlexUrrutia** [comment]
> There is no dedicated interface provided. You can access the image using VideoEffect. https://shogo4405.github.io/HaishinKit.swift/documentation/haishinkit/videoeffect

I forgot to mention I'm using the Flutter library, is it available?

**Reply by shogo4405** [reply]
Unfortunately, it's not available.

**Reply by AlexUrrutia** [reply]
ok  thanks for the quick response!
